### PR TITLE
Travis

### PR DESCRIPTION
**Pull request by Williangalvani**
Implemented Travis support.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+language: c
+before_install:
+  - mkdir Exposer
+  - mv *.cpp Exposer/
+  - mv *.h Exposer/
+  - mv *.ino Exposer/
```
Comment (patrickelectric) [inline .travis.yml]: mv -rf ./*!(Exposer) ./Exposer

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - sudo apt-get install -y freeglut3 freeglut3-dev 
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_1.pid
+--make-pidfile --background --exec /usr/bin/Xvfb -- :1 -ac -screen 0 1280x1024x16"
+  - sleep 3
+  - export DISPLAY=:1.0
+  - wget http://downloads.arduino.cc/arduino-1.6.9-linux64.tar.xz
```
Comment (patrickelectric) [inline .travis.yml]: get 1.6.10

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  - tar xf arduino-1.6.9-linux64.tar.xz
+  - sudo mv arduino-1.6.9 /usr/local/share/arduino
+  - sudo ln -s /usr/local/share/arduino/arduino /usr/local/bin/arduino
+  - sudo apt-get install libelf-dev gcc-avr binutils-avr gdb-avr avr-libc avrdude
+  - ls
```
Comment (patrickelectric) [inline .travis.yml]: remove

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  - sudo sh .travisScripts/get_simavr.sh
+
+script:
+  - arduino --board arduino:avr:nano:cpu=atmega328 --pref build.path=$PWD/Exposer/build --verify $PWD/Exposer/Exposer.ino
+  - ls
```
Comment (patrickelectric) [inline .travis.yml]: remove

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  - sudo sh .travisScripts/start_simavr.sh
+  - python --version
```
Comment (patrickelectric) [inline .travis.yml]: move this line to the last line of before install, and add the version of arduino

File: .travisScripts/get_simavr.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+root=$PWD
+#requirements
+
+git clone https://github.com/buserror/simavr.git
+
+cd simavr
+make
```
Comment (patrickelectric) [inline .travisScripts/get_simavr.sh]: make -C ./simavr

File: .travisScripts/start_simavr.sh [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+root=$PWD
+rm $root/simavr/examples/board_simduino/ATmegaBOOT_168_atmega328.ihex
```
Comment (patrickelectric) [inline .travisScripts/start_simavr.sh]: why this ?

File: .travisScripts/start_simavr.sh [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+mv $root/Exposer/build/Exposer.ino.hex  $root/simavr/examples/board_simduino/ATmegaBOOT_168_atmega328.ihex
```
Comment (patrickelectric) [inline .travisScripts/start_simavr.sh]: WHAT

File: .travis.yml (hunk 1 above)
Comment (Williangalvani) [inline .travis.yml]: I believe I found a better option, and it's alive!

File: .travis.yml (hunk 2 above)
Comment (Williangalvani) [inline .travis.yml]: done

File: .travis.yml (hunk 3 above)
Comment (Williangalvani) [inline .travis.yml]: ok

File: .travis.yml (hunk 4 above)
Comment (Williangalvani) [inline .travis.yml]: ok

File: .travisScripts/start_simavr.sh (hunk 7 above)
Comment (Williangalvani) [inline .travisScripts/start_simavr.sh]: I'm using the "Board_simduino" example to run the tests, it setups a predefined board with serial already out, so there's no need to change anything but the hex file. here I remove the default hex (a bootloader)

File: .travisScripts/start_simavr.sh (hunk 8 above)
Comment (Williangalvani) [inline .travisScripts/start_simavr.sh]: And here I add the new one

File: .travisScripts/get_simavr.sh (hunk 6 above)
Comment (Williangalvani) [inline .travisScripts/get_simavr.sh]: done

File: .travis.yml (hunk 5 above)
Comment (Williangalvani) [inline .travis.yml]: done, removed it

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  - sudo mv arduino-1.6.10 /usr/local/share/arduino
+  - sudo ln -s /usr/local/share/arduino/arduino /usr/local/bin/arduino
+  - sudo apt-get install libelf-dev gcc-avr binutils-avr gdb-avr avr-libc avrdude
+  - sudo sh .travisScripts/get_simavr.sh
+
+script:
+  - arduino --board arduino:avr:nano:cpu=atmega328 --pref build.path=$PWD/Exposer/build --verify $PWD/Exposer/Exposer.ino
+  - sudo sh .travisScripts/start_simavr.sh
+  - sudo python Exposer/tester.py /tmp/simavr-uart0
+  - sleep 10
```
Comment (patrickelectric) [inline .travis.yml]: remove this

File: .travisScripts/get_simavr.sh [hunk 10]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+root=$PWD
+#requirements
+
+git clone https://github.com/buserror/simavr.git
```
Comment (patrickelectric) [inline .travisScripts/get_simavr.sh]: add this as a submodule of the project use this script to manipulate

Comment (Williangalvani) [inline .travisScripts/get_simavr.sh]: done

**Pull request merged**